File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> answer;
    
    if (n > s) {
        answer.push_back(-1);
        
        return answer;
    }
    
    int div = s / n;

    for (int i = 1; i <= n; i++) {
        answer.push_back(div);
    }
    
    int rest = s % n;
    
    for (int i = 0; i < rest; i++) {
        answer[i]++;
    }
    
    sort(answer.begin(), answer.end());
    
    return answer;
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.*;

class Solution {
    public int[] solution(int n, int s) {
        int[] answer = new int[n];
        
        if (s < n) {
            int[] result = {-1};
            return result;
        }
        
        Arrays.fill(answer, s / n);
        
        for (int i = 0; i < s % n; i++) {
            answer[i]++;
        }
        
        Arrays.sort(answer);
        
        return answer;
    }
}